src/linkediting.js

```diff
@@ -55,5 +60,90 @@ export default class LinkEditing extends Plugin {
 
 		// Enable two-step caret movement for `linkHref` attribute.
 		bindTwoStepCaretToAttribute( editor.editing.view, editor.model, this, 'linkHref' );
+
+		// Setup highlight over selected link.
+		this._setupLinkHighlight();
+	}
+
+	/**
+	 * Adds highlight over link which has selection inside, together with two-step caret movement indicates whenever
+	 * user is typing inside the link.
+	 *
+	 * @private
+	 */
+	_setupLinkHighlight() {
+		const editor = this.editor;
+		const model = this.editor.model;
+		const doc = model.document;
+		const highlightDescriptor = {
+			id: 'linkBoundaries',
+			class: 'ck-link_selected',
+			priority: 1
+		};
+
+		// Convert linkBoundaries marker to view highlight.
+		editor.conversion.for( 'editingDowncast' )
+			.add( downcastMarkerToHighlight( {
+				model: 'linkBoundaries',
+				view: highlightDescriptor
+			} ) );
+
+		// Create marker over whole link when selection has "linkHref" attribute.
+		doc.on( 'change', () => {
```

Comment:
This solution will create an undo step. Have you tested how undo works with this solution?